text/0032-dust-protection/0032-dust-protection.md

```diff
@@ -0,0 +1,179 @@
+
+When an address is supposed to receive micro transactions, it must have an unspent output of a special type as a deposit. This deposit cannot be consumed by any transaction as long as the dust outputs remain unspent.
+
+An additional benefit of this rule is that it makes a mass of privacy violating [forced address reuse attacks](https://en.bitcoin.it/wiki/Privacy#Forced_address_reuse) more expensive to carry out.
+
+
+# Detailed design
+
+### Definitions
+
+*Dust output*: A transaction output that has an amount smaller than 1 Mi
+
+*SigLockedDustAllowanceOutput*: A new output type for deposits that enables an address to receive dust outputs. It can be consumed as an input like a regular `SigLockedSingleOutputs`.
+
+<table>
+  <tr>
+    <td><b>Name</b></td>
+    <td><b>Type</b></td>
+    <td><b>Description</b></td>
+  </tr>
+  <tr>
+    <td>Output Type</td>
+    <td>uint8</td>
+    <td>
+      Set to <strong>value 1</strong> to denote a <i>SigLockedDustAllowanceOutput</i>.
+    </td>
+  </tr>
+  <tr>
+    <td valign="top">Address</td>
+    <td colspan="2">
+      <details>
+        <summary>Ed25519 Address</summary>
+        <table>
+          <tr>
+            <td><b>Name</b></td>
+            <td><b>Type</b></td>
+            <td><b>Description</b></td>
+          </tr>
+          <tr>
+            <td>Address Type</td>
+            <td>uint8</td>
+            <td>
+              Set to <strong>value 1</strong> to denote an <i>Ed25519 Address</i>.
+            </td>
+          </tr>
+          <tr>
+            <td>Address</td>
+            <td>Array&lt;byte&gt;[32]</td>
+            <td>The raw bytes of the Ed25519 address which is a BLAKE2b-256 hash of the Ed25519 public key.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td>Amount</td>
+    <td>uint64</td>
+    <td>The amount of tokens to deposit with this <i>SigLockedDustAllowanceOutput</i> output.</td>
+  </tr>
+</table>
+
+
+### Validation
+
+Let A be the address that should hold the dust outputs' balances. Let S be the sum of all the amounts of all unspent `SigLockedDustAllowanceOutputs` on A. Then, the maximum number of allowed dust outputs on A is S divided by 10,000 and rounded down, i.e. 100 outputs for each 1 Mi deposited.
+
+The amount of a `SigLockedDustAllowanceOutput` must be at least 1 Mi. Apart from this, `SigLockedDustAllowanceOutputs` are processed identical to `SigLockedSingleOutput`. The transaction validation as defined in [Draft RFC-18](https://github.com/luca-moser/protocol-rfcs/blob/signed-tx-payload/text/0000-transaction-payload/0000-transaction-payload.md), however, needs to be adapted.
+
+_Syntactical validation_ for `SigLockedDustAllowanceOutput`:
+- The `Address` must be unique in the set of `SigLockedDustAllowanceOutputs` in one transaction T. However, there can be one `SigLockedSingleOutput` and one `SigLockedDustAllowanceOutputs` T.
```

Comment:
What's the motivation behind allowing a `SigLockedSingleOuput` + `SigLockedDustAllowanceOutput` within the same transaction targeting the same address?

Reply:
I think it makes more sense that the validation rules will be enforced per output type.

It can also be useful if someone wishes to reuse and address. Imagine someone having funds in an address. Now we want to split the funds so part of them will be the allowance without moving anything out of the address.

What would be the benefits of changing this rule?